### PR TITLE
[docs] Add description for EAS Build reference docs

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Follow-up for #20412

# How

This PR adds `description` to pages under the EAS Build > Reference section. Some minor tweaks and grammar fixes to match our writing style guide are also included.

# Test Plan

Changes have been tested locally.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/build-reference/android-builds.mdx [hunk 1]
```diff
@@ -17,7 +18,7 @@ The first phase happens on your computer. EAS CLI is in charge of completing the
 
    - Depending on the value of `builds.android.PROFILE_NAME.credentialsSource`, the credentials are obtained from either the local **credentials.json** file or from the EAS servers. If the `remote` mode is selected but no credentials exist yet, you're prompted to generate a new keystore.
 
-1. Create a tarball containing a copy of the repository. Actual behavior depends on the VCS workflow you are using. [Learn more here](https://expo.fyi/eas-vcs-workflow).
+1. Create a tarball containing a copy of the repository. Actual behavior depends on the[ VCS workflow](https://expo.fyi/eas-vcs-workflow) you are using.
```
Comment (Simek) [inline docs/pages/build-reference/android-builds.mdx:21]:
```suggestion
1. Create a tarball containing a copy of the repository. Actual behavior depends on the [VCS workflow](https://expo.fyi/eas-vcs-workflow) you are using.
```

Whitespace fix.

File: docs/pages/build-reference/e2e-tests.mdx [hunk 2]
```diff
@@ -499,20 +500,19 @@ const { openApp } = require('./utils/openApp');
...
-    await expect(element(by.text('Hi!'))).toBeVisible();
-  });
+it('shows "Hi!" after tapping "Click me"', async () => {
+await element(by.id('click-me-button')).tap();
+await expect(element(by.text('Hi!'))).toBeVisible();
+});
 });
-```
+
+````
```
Comment (Simek) [inline docs/pages/build-reference/e2e-tests.mdx:515]: Looks like unwanted reformat and change to four backticks at the end. Probably we need to add Prettier ignore there?

File: docs/pages/build-reference/e2e-tests.mdx [hunk 3]
```diff
@@ -573,7 +573,7 @@ const getLatestUpdateUrl = () =>
 const getAppId = () => appConfig?.expo?.extra?.eas?.projectId ?? '';
 
 const sleep = t => new Promise(res => setTimeout(res, t));
-```
+````
```
Comment (Simek) [inline docs/pages/build-reference/e2e-tests.mdx:576]: Leftover from issue above.

File: docs/pages/build-reference/ios-capabilities.mdx [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
+import { Terminal } from '~/ui/components/Snippet';
 
 When you make a change to your iOS entitlements, this change needs to be updated remotely on Apple's servers before making a production build. EAS Build automatically synchronizes capabilities on the Apple Developer Console with your local entitlements configuration when you run `eas build`. Capabilities are web services provided by Apple, think of them like AWS or Firebase services.
 
 > This feature can be disabled with `EXPO_NO_CAPABILITY_SYNC=1 eas build`
 
 ## Entitlements
 
-In bare workflow apps, the entitlements are read from your `ios/**/*.entitlements` file.
+In bare workflow apps, the entitlements are read from your **ios/.entitlements** file.
```
Comment (Simek) [inline docs/pages/build-reference/ios-capabilities.mdx:15]:
```suggestion
In bare workflow apps, the entitlements are read from your __ios/**/*.entitlements__ file.
```

I believe that glob for the file here was correct, but maybe this is a some kind of side-effect of switching to emphasized text via `**`. Alterative notation for bold text is `__` which would be better in this case.

**Review by Simek — APPROVED** [review]
Thank you for improving the docs content, LGTM! :+1:

Few small suggestions, mainly related to the formatting are below.

File: docs/pages/build-reference/e2e-tests.mdx (hunk 2 above)
Comment (amandeepmittal) [reply docs/pages/build-reference/e2e-tests.mdx:515]: Ah yes, missed this one and yes we need to add prettier ignore here. Will update it. Thanks for catching this and the other ones as well 👍

File: docs/pages/build-reference/ios-capabilities.mdx (hunk 4 above)
Comment (amandeepmittal) [reply docs/pages/build-reference/ios-capabilities.mdx:15]: You are right about the side-effect. I didn't notice this. Great catch!

**Pull request merged**